nemo/collections/tts/data/tts_dataset.py

```diff
@@ -248,6 +258,19 @@ def __init__(
                             text = self.text_normalizer_call(text, **self.text_normalizer_call_kwargs)
                         file_info["normalized_text"] = text
 
+                    # Check whether the user passed in a fixed vocab
+                    # If so, entries with illegal graphemes should be filtered out
+                    if getattr(self.text_tokenizer, "set_fixed_vocab", False):
+                        normalized_text = file_info["normalized_text"]
+                        text_set = set(set_grapheme_case(normalized_text, self.text_tokenizer.g2p.grapheme_case))
+                        # Skip if set difference between graphemes in text and valid tokens set is nonempty
+                        set_diff = text_set - tokens_set
+                        if set_diff:
+                            logging.warning(
+                                f"Skipping entry, found illegal grapheme(s) [{set_diff}] in text: [{normalized_text}]"
+                            )
```

Comment:
Right now it is possible for users to have characters in their training dataset that are not in their dictionary. At which point the tokenizer prints a warning that there is an unknown symbol. Why does the logic need to be different for the fixed vocab scenario?

Reply:
Good point--I'll remove the filtering code here.